File: TaskManager.Application/Helpers/QueryParams/TaskQueryParams.cs
using System;
using TaskManager.Domain.Enums;

namespace TaskManager.Application.Helpers.QueryParams;

public class TaskQueryParams : PaginationQueryParams
{
    public int? UserId { get; set; }
    public ProgressStatus? Status { get; set; }
    public ProgressStatus? ExcludeStatus { get; set; }
    public string OrderBy { get; set; } = "createdAt_desc";
}

File: TaskManager.Client/src/app/models/TaskQueryParams.ts
import { ProgressStatus } from "../_helpers/enums/progressStatus";
import { User } from "./User";

export class TaskParams{
    userId?: string;
    status?: ProgressStatus ;
    excludeStatus? :ProgressStatus;
    pageNumber = 1;
    pageSize = 15;
    orderBy = 'createdAt_desc'

    constructor(user: User | null){
        this.userId = user?.id;
    }
}
